### PR TITLE
refactor(jellyfish-api-core): move vault from loan into its own file

### PR DESCRIPTION
**Pull request by cwkang1998**
#### What this PR does / why we need it:

Part of #811 and #1145, this specifically seperates loan and vault in the `jellyfish-api-core` package.

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

**Review by ivan-zynesis — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts [hunk 1]
```diff
@@ -72,13 +72,13 @@ describe('Loan getInterest', () => {
     await testing.generate(1)
 
     const vaultAddress = await testing.generateAddress()
-    vaultId = await testing.rpc.loan.createVault({
+    vaultId = await testing.rpc.vault.createVault({
```
Comment (ivan-zynesis) [inline packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts:75]: Revert changes in `loan` test

Comment (cwkang1998) [reply packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts]: Got it, will only change it after full deprecation.

**Review by ivan-zynesis — APPROVED** [review]

File: packages/jellyfish-api-core/__tests__/category/vault/closeVault.test.ts [hunk 2]
```diff
@@ -0,0 +1,397 @@
+import { GenesisKeys, MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import BigNumber from 'bignumber.js'
+import { TestingGroup } from '@defichain/jellyfish-testing'
+
+describe('Loan', () => {
```
Comment (ivan-zynesis) [inline packages/jellyfish-api-core/__tests__/category/vault/closeVault.test.ts:5]:
```suggestion
describe('Vault', () => {
```

File: packages/jellyfish-api-core/__tests__/category/vault/createVault.test.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
+import BigNumber from 'bignumber.js'
+import { Testing } from '@defichain/jellyfish-testing'
+import { GenesisKeys, MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { VaultState } from '../../../src/category/vault'
+
+describe('Loan createVault', () => {
```
Comment (ivan-zynesis) [inline packages/jellyfish-api-core/__tests__/category/vault/createVault.test.ts:6]:
```suggestion
describe('Vault createVault', () => {
```

**Review by ivan-zynesis — COMMENTED** [review]
Test description

**Review by ivan-zynesis — APPROVED** [review]

**Review by canonbrother — DISMISSED** [review]

**Review by ivan-zynesis — APPROVED** [review]

**Review by chee-chyuan — APPROVED** [review]

**Review by canonbrother — APPROVED** [review]

**Pull request merged**